routers/api/actions/artifacts_chunks.go

```diff
@@ -67,9 +70,12 @@ type chunkFileItem struct {
 func listChunksByRunID(st storage.ObjectStorage, runID int64) (map[int64][]*chunkFileItem, error) {
 	storageDir := fmt.Sprintf("tmp%d", runID)
 	var chunks []*chunkFileItem
-	if err := st.IterateObjects(storageDir, func(path string, obj storage.Object) error {
-		item := chunkFileItem{Path: path}
-		if _, err := fmt.Sscanf(path, filepath.Join(storageDir, "%d-%d-%d.chunk"), &item.ArtifactID, &item.Start, &item.End); err != nil {
+	if err := st.IterateObjects(storageDir, func(fpath string, obj storage.Object) error {
+		baseName := filepath.Base(fpath)
+		// when read chunks from storage, it only contains storage dir and basename,
+		// no matter the subdirectory setting in storage config
+		item := chunkFileItem{Path: path.Join(storageDir, baseName)}
```

Comment:
It confused me that it sometimes calls `filepath` (line 74) while sometimes calls `path` (here). It looks quite fragile to me.

At least, IMO it needs to normalize the "paths" before using it.

Reply:
The second `path.Join` is not necessary. It saves chunk as filename `tmp/xxx.chunk` in above. So just `storageDir +"/" + basename` is enough.

Reply:
I still have difficulty to understand whether it is right to do these magic path operations.

Could it clarify these "path kinds"? If a path is a filesystem path, then it needs `filepath` package, if it is a web path (or a web path converted from filepath), then it needs `util.PathJoinRelX`.

Reply:
`baseName := filepath.Base(fpath)` is the real fix. `fpath` is from Storage and  can be filesystem path so use `filepath` to get basename.  

`chunkFileItem.Path` is a key to read and write file for Storage. When saving chunk, it uses `fmt.Sprintf("tmp%d/%d-%d-%d-%d.chunk", runID, runID, artifact.ID, start, end)` with `/` slash. 

Using `storageDir +"/" + basename` is enough in item.Path to read from Storage. Storage module can handle os difference.